src/opcode/mod.rs

```diff
@@ -38,6 +38,31 @@ pub enum Error {
     PushSize(Option<usize>),
 }
 
+/// Definitions needed for evaluation of script types.
+pub trait Evaluable {
+    /// The length in bytes of this script value. This can be more efficient than
+    /// `self.to_bytes().len()`.
+    fn byte_len(&self) -> usize;
+
+    /// Convert a script value into the bytes that would be included in a transaction.
+    fn to_bytes(&self) -> Vec<u8>;
+
+    fn restrict(pb: PossiblyBad) -> Result<Self, script::Error>
```

Comment:
This either needs documentation, or (more likely) should be:
```suggestion
    #[doc(hidden)]
    fn restrict(pb: PossiblyBad) -> Result<Self, script::Error>
```
(because `PossiblyBad` is not nameable in the public API).

Reply:
It's not obvious to me what this method does, so I think it should be documented whether or not it is usable in the public API. (Note that doc comments are allowed on an item with `#[doc(hidden)]`. Indeed there is an unstable `--document-hidden-items` flag to Rustdoc that shows them, although it is [currently not exposed by `cargo doc`](https://github.com/rust-lang/cargo/issues/7766#issuecomment-803731540).)